lib/battle_boats/engine.rb

```diff
@@ -27,21 +29,45 @@ def place_ships_manually
 
     def start
       interface.greet
-      until enemy_board.game_over?
+      until game_over?
```

Comment:
This method is responsible for **a lot**. Are there patterns here that could be extracted into their own methods? Or is this code simply shameless?

Reply:
I think there are concepts that could potentially be extracted here. To think through them, I like to think through the concepts I'm trying to model in the real world. In this case, I think you are working with the `Rules` of the `Game` and individual `Turn`s.